test/nestedJoin.integration.test.ts

```diff
@@ -0,0 +1,528 @@
+                    id: '0x08775ccb6674d6bdceb0797c364c2653ed84f3840002000000000000000004f0',
+                    address: '0x08775ccb6674d6bdceb0797c364c2653ed84f384',
+                    type: 'Weighted',
+                    level: 1,
+                    tokens: [
+                        {
+                            address:
+                                '0x79c58f70905f734641735bc61e45c19dd9ad60bc', // 3POOL-BPT
+                            decimals: 18,
+                            index: 0,
+                        },
+                        {
+                            address:
+                                '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2', // WETH
+                            decimals: 18,
+                            index: 1,
+                        },
+                    ],
+                },
+                {
+                    id: '0x79c58f70905f734641735bc61e45c19dd9ad60bc0000000000000000000004e7',
+                    address: '0x79c58f70905f734641735bc61e45c19dd9ad60bc',
+                    type: 'ComposableStable',
+                    level: 0,
+                    tokens: [
+                        {
+                            address:
+                                '0x6b175474e89094c44da98b954eedeac495271d0f', // DAI
+                            decimals: 18,
+                            index: 0,
+                        },
+                        {
+                            address:
+                                '0x79c58f70905f734641735bc61e45c19dd9ad60bc', // 3POOL-BPT
+                            decimals: 18,
+                            index: 1,
+                        },
+                        {
+                            address:
+                                '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48', // USDC
+                            decimals: 6,
+                            index: 2,
+                        },
+                        {
+                            address:
+                                '0xdac17f958d2ee523a2206206994597c13d831ec7', // USDT
+                            decimals: 6,
+                            index: 3,
+                        },
+                    ],
+                },
+            ],
+            mainTokens: [
+                {
+                    address: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2', // WETH
+                    decimals: 18,
+                },
+                {
+                    address: '0x6b175474e89094c44da98b954eedeac495271d0f', // DAI
+                    decimals: 18,
+                },
+                {
+                    address: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48', // USDC
+                    decimals: 6,
+                },
+                {
+                    address: '0xdac17f958d2ee523a2206206994597c13d831ec7', // USDT
+                    decimals: 6,
+                },
+            ],
+        };
+    }
+}
+
+export const grantRoles = async (
```

Comment:
Same as above 👍

Reply:
No need to refactor if we will remove it after relayer is deployed. 
Might be worth keeping this snippet in Notion though as it could be useful in future.

Reply:
I moved them to a relayerHelper file. I fear that they will be lost within Notion. In the source code, they are easier to find.
If our package is tree-shakeable and these methods are never used in production, they should not affect the b-sdk size anyway.
Let me know if you still think they should be deleted once they are no longer needed for the current tests.